iceoryx_posh/CMakeLists.txt

```diff
@@ -152,14 +152,15 @@ target_include_directories(iceoryx_posh
     source/version
     source/runtime
 )
-target_link_libraries(iceoryx_posh
+target_link_libraries(iceoryx_posh 
     PUBLIC
     iceoryx_utils::iceoryx_utils
+    ${ICEORYX_SANITIZER_FLAGS}        
```

Comment:
can't this be private?

Reply:
I have set  `ICEORYX_SANITIZER_FLAGS` as PRIVATE for all the used targets.
I am not sure what exactly are the implications.
From [cmake documentation](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) 

>  Libraries and targets following PRIVATE are linked to, but are not made part of the link interface

May be the dependent libraries dont carry over this flag, by marking it PRIVATE

Reply:
yes, if you use iceoryx_posh in your application and want to use the sanitizer, you need to set the flags yourself. Usually only the libraries should be public which are somehow exposed by the public API, e.g. if a posix header is included in a public header, we should also use PUBLIC in the link interface, else the user needs to add it herself, even if she uses that lib only by including our header. If the library is not exposed to the public API, then the link interface should be PRIVATE